R/generate_data.R

```diff
@@ -137,28 +137,29 @@ genData <- function(n, dtDefs = NULL, id = "id", envir = parent.frame()) {
 genDummy <- function(dtName, varname, sep = ".", replace = FALSE) {
 
   # Initial data checks
-
-  if (missing(dtName)) stop("argument 'dtName' is missing", call. = FALSE)
-  if (missing(varname)) stop("argument 'varname' is missing", call. = FALSE)
+  
+  assertNotMissing(dtName = missing(dtName), varname = missing(varname))
 
   # Check if data table exists
 
   if (!exists(deparse(substitute(dtName)), envir = parent.frame())) {
```

Comment:
Maybe we should create `assertDataTableExists` function. Another way we've checked the data table ([addColumns](https://github.com/kgoldfeld/simstudy/blob/403a6adadf6bb34bf396a68b4ab622a5c3fcd1b2/R/add_data.R#L27), for example) is by using `assertClass`, but this doesn't seem quite right, because it doesn't give the precise reason for error failure.

Reply:
That was my initial thought when making this, but I had some trouble trying to construct a function because if I tried to pass a dtName that didn't exist to the assert function, it would automatically return an error and wouldn't allow the assert to run. I'll take another look to see if I can figure something out.

Reply:
I haven't been able to solve this problem, and the workaround that I did sort of find doesn't seem to work either